tools/build_api.py

```diff
@@ -1348,11 +1348,13 @@ def merge_build_data(filename, toolchain_report, app_type):
             for project in tc.values():
                 for build in project:
                     try:
+                        build[0]['bin_fullpath'] = build[0]['bin']
+                        build[0]['elf_fullpath'] = build[0]['elf']
                         build[0]['elf'] = relpath(build[0]['elf'], path_to_file)
                         build[0]['bin'] = relpath(build[0]['bin'], path_to_file)
                     except KeyError:
                         pass
                     if 'type' not in build[0]:
                         build[0]['type'] = app_type
-                    build_data['builds'].append(build[0])
-    dump(build_data, open(filename, "w"), indent=4, separators=(',', ': '))
+                    build_data['builds'].insert(0, build[0])
```

Comment:
The build_data.json is not cleared and it will contain also failed builds. Previously the latest build was on the bottom of the build_data.json. This appends the build_data from the start so the latest build is in the top of the build_data.json. The run_icetea.py is looking this file to get all the test application binary paths and apparently read only the first build result. Other option to get the latest build is to start looking for the build date and which one is the latest, not sure if its reliable in every case. The date seems to be coming from python datetime method. (datetime.utcnow().isoformat())

Reply:
Sorry for being slow, can you explain this sentence a bit more?

> The run_icetea.py is looking this file to get all the test application binary paths and apparently read only the first build result.

Just curious why the order is important in this particular case, this isn't a blocker issue though.

Reply:
Previously if your first build failed, the failed build data was in the top of build_data.json, if the run_icetea.py only read this failed one and tried to get the test application binary paths from this, all the tests would fail because there was no binary path found. This would also happen in the future test runs if you don't use -c or delete the build_data.json separately.
Now the script also checks that was the build successful or not.

Reply:
Ok I see, sounds like a workaround then. Glad to hear the updated version checks the result! Thanks for the explanation